### PR TITLE
Adds a rudimentary 'demo mode' to the app

### PR DESCRIPTION
**Pull request by jessex**
For debugging and demo purposes. Pulls data from static files structured in the same way as live metric files. Data has been more or less randomized with some trends baked in for visual interest and testability. We'll need to keep this directory up to date as we add new files but we'd need to do this eventually anyway when we introduce unit testing to the app.

### CODE REVIEW

File: server/core/metricsApi.js [hunk 1]
```diff
@@ -114,20 +147,20 @@ function convertDownloadToJson(contents) {
   return jsonObject;
 }
 
-function fetchSnapshotMetrics(callback) {
-  return fetchMetrics('US_ND', 'snapshot', callback);
+function fetchSnapshotMetrics(isDemo, callback) {
+  return fetchMetrics('US_ND', 'snapshot', IS_DEMO, callback);
```
Comment (jjdressel11) [inline server/core/metricsApi.js]: Why are you having all of these functions get `isDemo` if you are using the environment `IS_DEMO` variable in the return calls?

File: server/core/metricsApi.js [hunk 2]
```diff
@@ -74,14 +94,27 @@ function fetchMetricsFromGCS(stateCode, metricType) {
  * First checks the cache to see if the metrics with the given type are already in memory and not
  * expired beyond the configured TTL. If not, then fetches the metrics for that type from the
  * appropriate files and invokes the callback only once all files have been retrieved.
+ *
+ * If we are in demo mode, then fetches the files from a static directory, /server/core/demoData/.
+ * Otherwise, fetches from Google Cloud Storage.
  */
-function fetchMetrics(stateCode, metricType, callback) {
+function fetchMetrics(stateCode, metricType, isDemo, callback) {
   return memoryCache.wrap(metricType, function (cacheCb) {
```
Comment (jjdressel11) [inline server/core/metricsApi.js]: Change to: 
```
  return memoryCache.wrap(metricType, (cacheCb) => {
    let fetcher = null;
    let source = null;
    if (isDemo) {
      source = 'local';
      fetcher = fetchMetricsFromLocal;
    } else {
      source = 'GCS';
      fetcher = fetchMetricsFromGCS;
    }
```

To fix a bunch of eslint errors.

File: src/components/charts/reincarcerations/ReincarcerationRateByReleaseFacility.js [hunk 3]
```diff
@@ -13,11 +13,18 @@ const ReincarcerationRateByReleaseFacility = (props) => {
...
+        recidivism_rate: recidivismRate,
+        state_code: stateCode,
+      } = data;
+      dataPoints.push([facility, recidivismRate, stateCode]);
     });
 
-    dataPoints = filterFacilities(dataPoints, 'RELEASE', 'US_ND');
+    if (dataPoints.length > 0) {
+      const stateCode = dataPoints[0][2];
+      dataPoints = filterFacilities(dataPoints, 'RELEASE', stateCode);
```
Comment (jjdressel11) [inline src/components/charts/reincarcerations/ReincarcerationRateByReleaseFacility.js]: This is great - thank you!

File: server/core/metricsApi.js (hunk 2 above)
Comment (jjdressel11) [reply server/core/metricsApi.js]: There are a bunch of existing eslint errors here that you could clean up if you wanted to 😎

Comment (jessex) [reply server/core/metricsApi.js]: I'm planning to submit a follow-up PR with all of the lint fixes!

File: server/core/metricsApi.js (hunk 1 above)
Comment (jessex) [reply server/core/metricsApi.js]: Ah that was a mistake. I changed approaches at one point and forgot to clean this up.

**Comment by jessex** [comment]
Ready for a final pass, @jjdressel11.

**Pull request merged**